packages/itwin/quantity-formatting/src/components/quantityformat/FormatSetPanel.tsx

```diff
@@ -0,0 +1,164 @@
+
+import type { UnitSystemKey } from "@itwin/core-quantity";
+import type { FormatSet } from "@itwin/ecschema-metadata";
+/**
+ * Props for the FormatSetPanel component
+ * @internal
+ */
+interface FormatSetPanelProps {
+  formatSet?: FormatSet;
+  editable?: boolean;
+  onFormatSetChange?: (formatSet: FormatSet) => void;
+}
+
+/**
+ * A React component that displays and allows editing of format set properties.
+ * @beta
+ */
+export const FormatSetPanel: React.FC<FormatSetPanelProps> = ({ formatSet, editable = false, onFormatSetChange }) => {
+  const [label, setLabel] = React.useState(formatSet?.label || "");
+  const [description, setDescription] = React.useState("");
+  const [unitSystem, setUnitSystem] = React.useState<string>("metric");
+  const { translate } = useTranslation();
+
+  // Generate unique IDs for form elements
+  const labelInputId = React.useId();
+  const descriptionInputId = React.useId();
+  const unitSystemSelectId = React.useId();
+
+  // Unit system options
+  const unitSystemOptions = [
+    { value: "metric", label: translate("QuantityFormat:labels.unitSystemMetric") },
+    { value: "imperial", label: translate("QuantityFormat:labels.unitSystemImperial") },
+    { value: "usCustomary", label: translate("QuantityFormat:labels.unitSystemUsCustomary") },
+    { value: "usSurvey", label: translate("QuantityFormat:labels.unitSystemUsSurvey") },
+  ];
+
+  // Update local state when formatSet prop changes
+  React.useEffect(() => {
+    setLabel(formatSet?.label || "");
+    setDescription(formatSet?.description || "");
+    setUnitSystem(formatSet?.unitSystem || "metric");
+  }, [formatSet]);
+
+  const handleLabelChange = React.useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      const newLabel = event.target.value;
+      setLabel(newLabel);
+
+      if (editable && formatSet && onFormatSetChange) {
+        const updatedFormatSet: FormatSet = {
+          ...formatSet,
+          label: newLabel,
+        };
+        onFormatSetChange(updatedFormatSet);
+      }
+    },
+    [editable, formatSet, onFormatSetChange],
+  );
+
+  const handleDescriptionChange = React.useCallback((event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    const newDescription = event.target.value;
+    setDescription(newDescription);
+
+    if (editable && formatSet && onFormatSetChange) {
+        const updatedFormatSet: FormatSet = {
+          ...formatSet,
+          description: newDescription,
+        };
+        onFormatSetChange(updatedFormatSet);
+    }
+  }, [editable, formatSet, onFormatSetChange]);
+
+  const handleUnitSystemChange = React.useCallback((value: string) => {
+    setUnitSystem(value);
+
+    if (editable && formatSet && onFormatSetChange) {
+        const updatedFormatSet: FormatSet = {
+          ...formatSet,
+          unitSystem: value as UnitSystemKey,
+        };
+        onFormatSetChange(updatedFormatSet);
+    }
+  }, [editable, formatSet, onFormatSetChange]);
+
+  if (!formatSet) {
```

Comment:
Why does this component allows passing `undefined` `formatSet`? I think message about selecting format set should be rendered by the parent of this component.